### PR TITLE
optimize helm valuese generate

### PR DESCRIPTION
**Pull request by misstick86**
As I know, `helm.go` file  function is to replace the variables in the `values.tmpl.yaml` file with environment variables, and then generate `values.yaml` file. This function can be implemented using the `envsubst` command.

### CODE REVIEW

File: charts/gateway-helm/values.tmpl.yaml [hunk 1]
```diff
@@ -44,4 +44,4 @@ envoyGatewayMetricsService:
     protocol: TCP
     targetPort: https
 
-kubernetesClusterDomain: cluster.local
+kubernetesClusterDomain: cluster.local
```
Comment (zirain) [inline charts/gateway-helm/values.tmpl.yaml:47]: add new line at end.

**Review by zirain — COMMENTED** [review]
LGTM

**Comment by arkodg** [comment]
guessing this works for Mac and Linux but will fail for Windows ?

**Comment by zirain** [comment]
use https://github.com/drone/envsubst ?

File: tools/make/helm.mk [hunk 2]
```diff
@@ -32,4 +32,4 @@ helm-release: helm-generate
 	helm package charts/${CHART_NAME} --app-version ${TAG} --version ${CHART_VERSION} --destination ${OUTPUT_DIR}/charts/
 
 helm-generate:
-	IMAGE=${IMAGE} RELEASE_TAG=${TAG} go run helm/*
+	ImageRepository=${IMAGE} ImageTag=${TAG} envsubst < charts/gateway-helm/values.tmpl.yaml > ./charts/gateway-helm/values.yaml
```
Comment (Xunzhuo) [inline tools/make/helm.mk:35]: I am sure the Env Name is correct. When release Helm chart, we use IMAGE and RELEASE_TAG from release workflow. ImageRepository and ImageTag are not set in release workflow.

Comment (misstick86) [reply tools/make/helm.mk:35]: @Xunzhuo oh, I see what you mean. when helm chart releasing, you'll set a `TAG` environment variable. [release](https://github.com/envoyproxy/gateway/blob/main/.github/workflows/release.yaml#L34) but it's helm tag(may be it called helm app version), not *gateway* image tag. Should we set two different variable names to differentiate this?

Comment (Xunzhuo) [reply tools/make/helm.mk:35]: Plz add changes like shown above.

Comment (misstick86) [reply tools/make/helm.mk:35]: added, please check again.

File: charts/gateway-helm/values.tmpl.yaml (hunk 1 above)
Comment (misstick86) [reply charts/gateway-helm/values.tmpl.yaml:47]: Shall we need this? according to this [pr](https://github.com/envoyproxy/gateway/pull/1277/files), this is removed.

Comment (Xunzhuo) [reply charts/gateway-helm/values.tmpl.yaml:47]: We can remove it for now.

Comment (misstick86) [reply charts/gateway-helm/values.tmpl.yaml]: done

**Review by Xunzhuo — APPROVED** [review]
Looks good to me. Thanks

**Review by zirain — APPROVED** [review]

**Comment by misstick86** [comment]
there are a little bit weird.  It's total fine When I execute this testing method alone. Shall I do something about it now?

**Pull request merged**